src/Graph.elm

```diff
@@ -1301,16 +1285,9 @@ stronglyConnectedComponents graph =
 {-| Returns a string representation of the graph in the format of
 `Graph.fromNodesAndEdges [<nodes>] [<edges>]`.
 -}
-toString : Graph n e -> String
-toString graph =
-    let
-        nodeList =
-            nodes graph
-
-        edgeList =
-            edges graph
-    in
+toString : (n -> String) -> (e -> String) -> Graph n e -> String
```

Comment:
It would be nice to have couple of unit tests for this more complex toString implementation.

Reply:
What kind of unit tests do you have in mind?  The only difference is that you have to pass in your own converters.  I did use this for the test error messages, so even though there aren't specific tests for it, it is compiling.  I'll write a simple test to just prove it works correctly.